File: booking-app/src/app/models/comment.ts
import { Accommodation } from './accommodation';
import { User } from './user';

export class Comment {
  constructor(
    public Id: number,
    public Grade: number,
    public Text: string,
    public AppUser: User,
    public AppUserId: number,
    public Accommodation: Accommodation,
    public AccommodationId: number
){}
}

File: BookingApp/Models/Data/Comment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace BookingApp.Models.Data
{
    public class Comment
    {
        public int Id { get; set; }

        public String Text { get; set; }

        [Required]
        [Range(1, 10)]
        public double Grade { get; set; }

        [Key]
        [Column(Order = 1)]
        [ForeignKey("AppUser")]
        public int AppUserId { get; set; }
        public AppUser AppUser { get; set; }

        [Key]
        [Column(Order = 2)]
        [ForeignKey("Accommodation")]
        public int AccommodationId { get; set; }
        public Accommodation Accommodation { get; set; }

    }
}